src/subdomains/user/application/services/kyc.service.ts

```diff
@@ -0,0 +1,74 @@
+import { BadRequestException, Injectable, ServiceUnavailableException } from '@nestjs/common';
+import { Method } from 'axios';
+import { CryptoService } from 'src/blockchain/crypto.service';
+import { Config } from 'src/config/config';
+import { HttpError, HttpService } from 'src/shared/services/http.service';
+import { User } from '../../domain/entities/user.entity';
+import { UserService } from './user.service';
+import { WalletService } from './wallet.service';
+
+@Injectable()
+export class KycService {
+  constructor(
+    private readonly http: HttpService,
+    private readonly cryptoService: CryptoService,
+    private readonly userService: UserService,
+    private readonly walletService: WalletService,
+  ) {}
+
+  async startKyc(address: string): Promise<User> {
+    const wallet = await this.walletService.getByAddress(address, true);
+    if (!wallet) throw new BadRequestException('Wallet not found');
+    const user = wallet.user;
+    if (!user) throw new BadRequestException('User not found');
+    if (user.kycHash) return user;
+    user.kycId = await this.cryptoService.getAddress(user.id);
+    const signature = await this.getSignature(user.id, user.kycId);
+    const accessToken = await this.signUp(user.kycId, signature);
+    const kycUser = await this.getUser(accessToken);
+
+    return await this.userService.updateUser(user.id, { kycHash: kycUser.kycHash, mail: kycUser.mail });
+  }
+
+  async signUp(address: string, signature: string): Promise<string> {
+    const { accessToken } = await this.callApi<{ accessToken }>('auth/signUp', 'POST', {
+      address,
+      signature,
+      walletId: 7,
+    });
+    return accessToken;
+  }
+
+  async getSignMessage(address: string): Promise<{ message: string }> {
+    return await this.callApi<{ message: string }>(`auth/signMessage?address=${address}`);
+  }
+
+  async getUser(accessToken: string): Promise<User> {
+    return await this.callApi<User>(`user`, 'GET', undefined, accessToken);
+  }
+  // --- HELPER METHODS --- //
+
+  async getSignature(userId: number, address: string): Promise<string> {
```

Comment:
I would use a different DTO here (because fields may be different at DFX), we just need the kycHash